airflow/api_fastapi/common/types.py

```diff
@@ -59,6 +66,20 @@ class TimeDelta(BaseModel):
 TimeDeltaWithValidation = Annotated[TimeDelta, BeforeValidator(_validate_timedelta_field)]
 
 
+def _validate_nonnaive_datetime_field(dt: datetime | None) -> datetime | None:
+    """Validate and return the datetime field."""
+    if dt is None:
+        return None
+    if isinstance(dt, str):
+        dt = datetime.fromisoformat(dt)
+    if not dt.tzinfo:
+        raise ValueError("Invalid datetime format, Naive datetime is disallowed")
+    return dt
+
+
+DatetimeWithNonNaiveValidation = Annotated[datetime, BeforeValidator(_validate_nonnaive_datetime_field)]
+
+
```

Comment:
Pydantic has a native `AwareDatetime` (we use it a little bit) that does exactly that I believe.

At some point we might want to switch all datetime to `AwareDatetime` I think

Reply:
Done, thanks!